app/ts/library/ethereum.ts

```diff
@@ -41,28 +41,18 @@ export function isTransferTransaction(txResponse: TransactionResponse): txRespon
 }
 
 export const erc20Interface = new Interface(ERC20ABI)
-
-export function parseLogArgsFromReceipt(transactionReceipt: TransactionReceipt) {
-	// workaround for https://github.com/ethers-io/ethers.js/issues/4029
-	const transferLog = transactionReceipt.logs.find(isTransferLog) as unknown
-	if (transferLog === undefined) return undefined
-	const logArgs = erc20Interface.parseLog(transferLog as { topics: string[]; data: string })?.args
-	return isTransferResult(logArgs) ? logArgs : undefined
-}
-
 export const transferTopic = id('Transfer(address,address,uint256)')
 
-export function isTransferLog(log: Log) {
-	const [topic] = log.topics
-	return topic === transferTopic
-}
-
-export interface TransferResult extends Result {
-	from: string
-	to: string
-	value: bigint
+export function extractArgValue<T>({ topics, data }: Log, argKey: string): T | null {
+	// ensure topics conforms to parseLog param type
+	const mutableTopics = topics.map(topic => topic)
+	const parsedLog = erc20Interface.parseLog({ topics: mutableTopics, data })
```

Comment:
This can be simplified (and I suspect is faster too, but that is insignificant).  You may still want a variable for `mutableTopics` to make it clear to the reader why you are making a copy of the array.
```suggestion
	const parsedLog = erc20Interface.parseLog({ topics: [...topics], data })
```
```suggestion
	const mutableTopics = [...topics]
	const parsedLog = erc20Interface.parseLog({ topics: mutableTopics, data })
```

Reply:
`erc20Interface.parseLog` will throw if it fails to parse logs. Eg the logs do not conform to erc20, I have this kind of ugly helper function to map the failures to null:
```ts
const parseLogIfPossible = (ethersInterface: ethers.Interface, log: { topics: string[], data: string}) => {
	try {
		return ethersInterface.parseLog(log)
	} catch (error) {
		return null
	}
}
```

Reply:
> This can be simplified (and I suspect is faster too, but that is insignificant). You may still want a variable for `mutableTopics` to make it clear to the reader why you are making a copy of the array.

i added a helper as @KillariDev mentioned and used the spread operator for topics to conform to parseLog parameters